dask/bag/tests/test_bag.py

```diff
@@ -300,6 +306,42 @@ def test_map_partitions():
     assert b.map_partitions(lambda a: len(a) + 1).name != b.map_partitions(len).name
 
 
+def test_map_partitions_with_kwargs():
+    corpus = '''
+    lorem ipsum dolor sit amet consectetur adipiscing elit suspendisse
+    sollicitudin molestie arcu ac rutrum quam fringilla in etiam luctus
+    consectetur neque in viverra etiam pharetra tristique fermentum mauris
+    non nulla mattis interdum risus nec faucibus quam praesent varius ex id
+    bibendum iaculis justo metus ornare dolor sit amet finibus ante metus et
+    eros vivamus mattis est eget ligula blandit sed tempor urna lacinia
+    maecenas convallis justo ut fringilla efficitur donec elementum sed nisi
+    et facilisis maecenas feugiat condimentum accumsan nulla facilisi
+    suspendisse convallis arcu vitae feugiat dictum donec tincidunt nisi in
+    cursus faucibus suspendisse quam mauris consequat quis finibus in
+    gravida a dui
+    '''
+    words = corpus.strip().split()
+    word_bag = db.from_sequence(words, npartitions=10)
+
+    def code_words(part, all_words=None):
+        codes = dict((word, code)
+                     for code, word in enumerate(sorted(all_words)))
+        return [codes[word] for word in part]
+
+    all_the_words = word_bag.reduction(
+        perpartition=lambda part: set(part),
+        aggregate=lambda sets: reduce(set.union, sets))
+    assert all_the_words.compute() == set(words)
+
+    coded_word_bag = word_bag.map_partitions(
+        code_words, all_words=all_the_words)
+
+    expected_code = dict((word, code)
+                         for code, word in enumerate(sorted(set(words))))
+    print(expected_code)
+    assert coded_word_bag.compute(get=dask.get) == [expected_code[word] for word in words]
```

Comment:
This unittest seems like it has some complexity that is unrelated to the code.  A future developer might find that they have to spend a minute or two understanding your application in order to properly debug a failing test in the future.  Is there a simpler set of examples we can test instead?  I find basic arithmetic to be fairly broadly accessible.